File: Cinema_groupe4_back/cinema_backend/src/main/java/com/intiformation/modele/Programmation.java
package com.intiformation.modele;

import java.util.Date;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Programmation {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id_programmation")
	private long idProgrammation;
	private Date dateHeureProg;
	@ManyToOne
	@JoinColumn(name = "film_id", referencedColumnName = "visaExploitation")
	private Film film;
	@ManyToOne
	@JoinColumn(name = "salle_id", referencedColumnName = "idSalle")
	private Salle salle;

	public Programmation() {
	}

	public Programmation(long numeroPlace, boolean isUsed, Date dateHeureProg, List<Place> listePlaces) {
		this.dateHeureProg = dateHeureProg;
	}

	public Programmation(long numeroPlace, boolean isUsed, Date dateHeureProg, List<Place> listePlaces, Film film,
			Salle salle) {
		this.dateHeureProg = dateHeureProg;
		this.film = film;
		this.salle = salle;
	}

	public long getIdProgrammation() {
		return idProgrammation;
	}

	public void setIdProgrammation(long idProgrammation) {
		this.idProgrammation = idProgrammation;
	}

	public Date getDateHeureProg() {
		return dateHeureProg;
	}

	public void setDateHeureProg(Date dateHeureProg) {
		this.dateHeureProg = dateHeureProg;
	}

	public Film getFilm() {
		return film;
	}

	public void setFilm(Film film) {
		this.film = film;
	}

	public Salle getSalle() {
		return salle;
	}

	public void setSalle(Salle salle) {
		this.salle = salle;
	}

}

File: Cinema_groupe4_front/cinema_front/src/app/model/programmation.ts
import { Film } from "./film";
import { Place } from "./place";
import { Salle } from "./salle";

export class Programmation{

    private _idProgrammation : number;
	private _dateHeureProg : Date;
	private _film : Film;
	private _salle : Salle;
	private _listePlaces :Place[];

    constructor(idProgrammation : number, dateHeureProg : Date, film : Film, salle : Salle, listePlaces :Place[]){
        this._idProgrammation = idProgrammation;
        this._dateHeureProg = dateHeureProg;
        this._film = film;
        this._salle = salle;
        this._listePlaces = listePlaces;
    }

    get idProgrammation() : number {
        return this._idProgrammation;
    }

    set idProgrammation(value : number) {
        this._idProgrammation = value;
    }

    get dateHeureProg() : Date {
        return this._dateHeureProg;
    }

    set dateHeureProg(value : Date) {
        this._dateHeureProg = value;
    }

    get film() : Film {
        return this._film;
    }

    set film(value : Film) {
        this._film = value;
    }

    get salle() : Salle {
        return this._salle;
    }

    set salle(value : Salle) {
        this._salle = value;
    }

    get listePlaces() : Place[] {
        return this._listePlaces;
    }

    set listePlaces(value : Place[]) {
        this._listePlaces = value;
    }

    
}